src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -636,24 +641,19 @@ class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long),
           throw new IAE(s"Unknown value type ${x.getClass} : [$x]")
       }
       val dateBucket = gran.truncate(new DateTime(k)).getMillis
-      val modSize = maxTimePerBucket.get(dateBucket.toLong)
-      if (modSize.isEmpty) {
-        // Lazy ISE creation
-        throw new ISE(s"bad date bucket [${dateBucket.toLong}]. " +
-          s"available: ${maxTimePerBucket.keySet}")
+      val shardLookup = shardLookups.get(dateBucket)
```

Comment:
Any time I tried that previously it always created the exception regardless of if the throw branch was actually taken. This method did not do so